bluemira/equilibria/flux_surfaces.py

```diff
@@ -32,8 +32,12 @@
 import matplotlib.pyplot as plt
 import numba as nb
 import numpy as np
+import pointpats
 from scipy.integrate import solve_ivp
+from scipy.special import lpmv
+from shapely.geometry import Polygon
```

Comment:
Would it be possible to avoid using shapely and pointspat here? We worked quite hard to rid of shapely, and we try to keep the number of dependencies to a minimum.. 

We have the `Coordinates` class, which is just a soft wrapper around a coordinate array, along with stuff like `in_polygon` to see if a point is inside a polygon. `bluemira/geometry/coordinates.py`

I can appreciate that generating a Poisson distribution of points inside a polygon is less trivial than just getting random points inside a polygon.

Reply:
I agree with not using those dependencies. Also it should be noted that new random number generation code should use the below pattern as per the numpy docs https://numpy.org/doc/stable/reference/random/index.html#random-quick-start:

```python
rng = np.random.default_rng()
rng.uniform()
```
Looking into the pointpats code it looks like they are just using uniform random numbers with low/high set to the (x,y) min/max see [here](https://github.com/pysal/pointpats/blob/9a09bd7ff2c3b5402f02a0a54175193cb643edd8/pointpats/random.py#L134). Alternatively you could seemingly just generate a set of random numbers in the default half open state with a given size and then unnormalise it using our [RegionInterpolator](https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/8651d947ef80438af4582182e3b37d0731bb27f7/bluemira/utilities/positioning.py#L132).

I may have missed something in their code however

Reply:
Shapely and pointspat removed. Possion distribution is not necessary, so was not an issue with changes. Thanks for the help.